### PR TITLE
Improvements for NXP MIMXRT1010-EVK

### PR DESCRIPTION
**Pull request by rexut**
- set pull-up option bit correct for "User Button"
- Arduino UNO R3 header: add GPIO map
- Arduino UNO R3 header: add I2C label
- Arduino UNO R3 header: add SPI label

### CODE REVIEW

**Review by de-nordic — APPROVED** [review]
OK for FAT FS sample

File: samples/subsys/fs/fat_fs/sample.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ tests:
     platform_allow: nrf52840_blip
   sample.filesystem.fat_fs.adafruit_2_8_tft_touch_v2:
     depends_on: arduino_spi arduino_gpio arduino_i2c
-    platform_exclude: stm32f769i_disco pan1781_evb pan1782_evb frdm_k64f
+    platform_exclude: stm32f769i_disco pan1781_evb pan1782_evb frdm_k64f mimxrt1010_evk
```
Comment (mmahadevan108) [inline samples/subsys/fs/fat_fs/sample.yaml:17]: WHy is fat_fs sample excluded?

Comment (rexut) [reply samples/subsys/fs/fat_fs/sample.yaml:17]: The sample requires the `adafruit_2_8_tft_touch_v2` shield and this shield requires an special Arduino header signal for the SPI CS (AFAICR signal D10). Sadly this signal is not connected from SOC to the header in factory default on this board. So the `gpio-map` will not provide the needed signal label for the Adafruit display shield and the build will fail. That's why this board exclusion is here (and all the others).

Comment (rexut) [reply samples/subsys/fs/fat_fs/sample.yaml:17]: @mmahadevan108 : Does that answer your question? If so, I'd like to close this conversation.

**Review by henrikbrixandersen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**